File: python/program/BarangEdisiTerbatas.py
from BarangPremium import BarangPremium

# Class Level 3: BarangEdisiTerbatasfrom BarangPremium import BarangPremium


class BarangEdisiTerbatas(BarangPremium):
    def __init__(self, id, nama, merek, harga, material, fitur_eksklusif, nomor_seri, jumlah_produksi):
        super().__init__(id, nama, merek, harga, material, fitur_eksklusif)
        self.nomor_seri = nomor_seri
        self.jumlah_produksi = jumlah_produksi


    def get_nomor_seri(self):
        return self.nomor_seri
    def get_jumlah_produksi(self):
        return self.jumlah_produksi

    def set_nomor_seri(self, nomor_seri):
        self.nomor_seri = nomor_seri
    def set_jumlah_produksi(self, jumlah_produksi):
        self.jumlah_produksi = jumlah_produksi

    def display(self, col_widths):
        return f"{super().display(col_widths)} {self.get_nomor_seri():<{col_widths['Nomor Seri']}} | {self.get_jumlah_produksi():<{col_widths['Jumlah Produksi']}} |"

File: java/program/BarangEdisiTerbatas.java

import java.util.Map;
import java.text.DecimalFormat;
// Class Level 3: BarangEdisiTerbatas
class BarangEdisiTerbatas extends BarangPremium {
    private String nomorSeri;
    private int jumlahProduksi;

    public BarangEdisiTerbatas(int id, String nama, String merek, double harga, String material, String fiturEksklusif, String nomorSeri, int jumlahProduksi) {
        super(id, nama, merek, harga, material, fiturEksklusif);
        this.nomorSeri = nomorSeri;
        this.jumlahProduksi = jumlahProduksi;
    }

    // Getters
    public String getNomorSeri() { return nomorSeri; }
    public int getJumlahProduksi() { return jumlahProduksi; }

    // Setters
    public void setNomorSeri(String nomorSeri) { this.nomorSeri = nomorSeri; }
    public void setJumlahProduksi(int jumlahProduksi) { this.jumlahProduksi = jumlahProduksi; }

    // fungsi di override biar display ga tumpang tindih
    @Override
    public void display(Map<String, Integer> colWidths) {
        super.display(colWidths);
        System.out.printf(" %-" + colWidths.get("Nomor Seri") + "s | %-" + colWidths.get("Jumlah Produksi") + "s |\n",  getNomorSeri(), getJumlahProduksi());
    }
}